### ISSUE TITLE
Video upload with tus (JavaScript File object)

### ISSUE THREAD

**Issue by manolismax** [vimeo/vimeo.js#69]
Hello,

according to the following link, 
https://help.vimeo.com/hc/en-us/articles/224819007-Can-my-upload-script-support-resumable-uploads-or-a-progress-bar-?flash_digest=ebcf64d9b1169b33257ebf61ec2293e31284a8cb

this library can support tus uploads.

I am following your example, but as i found 
**client.upload** method supports only relative path files from server as i suppose.

Although the **tus** protocol supports JavaScript File object.

I found on node_modules/vimeo/lib/vimeo.js the below code

```
  var file = fs.createReadStream(filePath)
  var upload = new tus.Upload(file, { ...
```
So is any workaround to bypass the path restriction and give file?

**Comment by erunion** [comment]
This library currently only supports uploading from files on the filesystem.

**Comment by invincible1388** [comment]
By any chance does the library support video blob uploads?
I am trying to upload video from webcam recorded video from front end.

**Comment by GeoSot** [comment]
You have to override  

> Vimeo.prototype.upload 

and 

> Vimeo.prototype._performTusUpload

or make two copies without fs streamer.


Specially :
on   Vimeo.prototype.upload omit the
```
  try {
    var fileSize = fs.statSync(filePath).size
  } catch (e) {
    return errorCallback('Unable to locate file to upload.')
  }
```
and on Vimeo.prototype._performTusUpload, omit
 `var file = fs.createReadStream(filePath)`

You need some more changes in order Vimeo.prototype.upload   to be full functional, but is a good start.

**Comment by erunion** [comment]
@invincible1388 Unfortunately right now, no. There are some updates I'm going to be making over the next week or two, I might be able to squeeze in stream and blob support.

**Comment by invincible1388** [comment]
Thanks for the comment. I did went through the library implementation and made some changes to support blob formats too. Now its working fine for me.

**Comment by erunion** [comment]
@invincible1388 Would you want to PR those changes?

**Comment by jakobwarrer** [comment]
Any update on this?

**Comment by erunion** [comment]
@jakobwarrer Unfortunately not on our end. If you're up for taking a stab at writing support for it into the library, we'd gladly merge it in.

**Comment by josephharveyangeles** [comment]
Duplicate of #80 A PR is on the way #86

**Comment by erunion** [comment]
Closing this in favor of #80.

**Comment by erunion** [comment]
This is now resolved in [v2.1.1](https://github.com/vimeo/vimeo.js/releases/tag/2.1.1).
